tests/src/main/java/net/neoforged/neoforge/debug/transfer/VanillaHandlersTests.java

```diff
@@ -0,0 +1,566 @@
+            if (resourceHandler.insert(book, 2, transaction) != 2) throw helper.assertionException("Should have inserted 2 books");
+
+            if (bookshelf.getItem(0).getCount() != 1) throw helper.assertionException("Bookshelf stack 0 should have size 1");
+            if (!book.matches(bookshelf.getItem(0))) throw helper.assertionException("Bookshelf stack 0 should be a book");
+            if (bookshelf.getItem(1).getCount() != 1) throw helper.assertionException("Bookshelf stack 1 should have size 1");
+            if (!book.matches(bookshelf.getItem(1))) throw helper.assertionException("Bookshelf stack 1 should be a book");
+        }
+
+        if (!bookshelf.getItem(0).isEmpty()) throw helper.assertionException("Bookshelf stack 0 should be empty again after aborting transaction");
+        if (!bookshelf.getItem(1).isEmpty()) throw helper.assertionException("Bookshelf stack 1 should be empty again after aborting transaction");
+
+        // Second, check that we correctly update the last modified slot.
+        try (Transaction tx = Transaction.open(null)) {
+            if (resourceHandler.insert(1, book, 1, tx) != 1) throw helper.assertionException("Should have inserted 1 book");
+            if (bookshelf.getLastInteractedSlot() != 1) throw helper.assertionException("Last modified slot should be 1");
+
+            if (resourceHandler.insert(2, book, 1, tx) != 1) throw helper.assertionException("Should have inserted 1 book");
+            if (bookshelf.getLastInteractedSlot() != 2) throw helper.assertionException("Last modified slot should be 2");
+
+            if (resourceHandler.extract(1, book, 1, tx) != 1) throw helper.assertionException("Should have extracted 1 book");
+            if (bookshelf.getLastInteractedSlot() != 1) throw helper.assertionException("Last modified slot should be 1");
+
+            // Now, create an aborted nested transaction.
+            try (Transaction nested = Transaction.open(tx)) {
+                if (resourceHandler.insert(book, 100, nested) != 5) throw helper.assertionException("Should have inserted 5 books");
+                // Now, last modified slot should be 5.
+                if (bookshelf.getLastInteractedSlot() != 5) throw helper.assertionException("Last modified slot should be 5");
+            }
+
+            // And it's back to 1 in theory.
+            if (bookshelf.getLastInteractedSlot() != 1) throw helper.assertionException("Last modified slot should be 1");
+            tx.commit();
+        }
+
+        if (bookshelf.getLastInteractedSlot() != 1) throw helper.assertionException("Last modified slot should be 1 after committing transaction");
+
+        // Let's also check the state properties. Only slot 2 should be occupied.
+        BlockState state = bookshelf.getBlockState();
+
+        if (state.getValue(BlockStateProperties.CHISELED_BOOKSHELF_SLOT_0_OCCUPIED)) throw helper.assertionException("Slot 0 should not be occupied");
```

Comment:
Can we loop `ChiseledBookShelfBlock.SLOT_OCCUPIED_PROPERTIES` instead of hardcoding all of these?